test/gtest/common/test.cc

```diff
@@ -70,13 +70,36 @@ void test_base::set_config(const std::string& config_str)
         value = config_str.substr(pos + 1);
     }
 
-    optional = false;
-    if ((name.length() > 0) && name.at(name.length() - 1) == '?') {
-        name = name.substr(0, name.length() - 1);
-        optional = true;
+    mode = FAIL_IF_NOT_EXIST;
+
+    /*
+     * What happens if NAME is not a valid configuration key?
+     * - "NAME=VALUE"   : fail the test
+     * - "NAME?=VALUE"  : set UCX_NAME environment variable
+     * - "NAME:=VALUE"  : skip the test
```

Comment:
maybe better `~=` or smth different?
IMO, `:=` means assignment and could be confusing then

Reply:
done